sfcasts/patterns.md

```diff
@@ -1,62 +1,36 @@
-
-Instead of putting all that code in one giant class. We'll talk about two behavioral
-patterns, the strategy pattern and the observer pattern. Okay, enough defining
-things. We are going to get technical, but we are also going to use these patterns in
-a real Symfony project to do real stuff. We'll only cover a few patterns in this
-tutorial, some of my favorites, but if you want to see more after, let us know. All
-right. To get the most out of this tutorial, as always, you should definitely
-download the code and the code along with me. Download the course code from this
-page. After you un zip it, you'll find a start directory that has the same code that
-you see here. Pop open this readme.md file. For the set of details though, this one's
-pretty easy. You just gotta run, composer, install. The app we have here is a simple
-command line R PPG, where characters battle each other and level up. It's like my
-favorite type of game to run it. Run bin console app, co game,:play. Okay, sweet. We
-have three character types. Let's be a fighter. We're fighting another fighter Q Epic
-Battle sand sounds, and we won 11 rounds of fighting 94 damage doubt, 84 damage
-received. And if we want, we can fight again and oh, we are on a roll.
-
-All right, so this is a Symfony app, but it's a very simple Symfony app. It's not
-even the first thing we have is this command class. I want you to understand a little
-bit about how the game, how the app is set up right now. This basically just sets
-things up and prints the results. It figures out which character you want to be and
-then it actually starts the battle. Um, but most of the work is actually done, is
-done here on this game object, which is this game application. This takes these two
-character objects and it actually does the logic of having them attack each other
-until somebody wins. It also at the bottom contains the three characters we have,
-which are represented by this character object and that character object. You can
-pass in different max health, different base damage that you do and different armor
-levels. So game application with care three characters here, and that battles them up
-up here, and that's basically it. So next, let's get into our first pattern, which is
-going to be this strategy pattern where we make our characters able to cast magical
-spells, which will require making this class a lot more flexible.
+A design pattern is nothing more than a strategy for writing your code when you have a particular problem. If you can start to identify which types of problems are solved by which strategies, you'll walk into situations and immediately know what to do. Learning these design patterns gives you:
+
+A) More tools in your developer toolkit when coding and
+B) A better understanding of core libraries like Symfony, which leverages design patterns frequently.
+
+It's pretty cool to look at core code and say
+
+`Oh, I see what you're doing there. That's the
+decorator pattern.`
```

Comment:
I changed this into a funny party reference - showing two simple cartoon people at a party chatting would be awesome :)